integration/yarn/src/main/java/tachyon/yarn/Client.java

```diff
@@ -0,0 +1,388 @@
+  /** Container context to launch application master */
+  private ContainerLaunchContext mAmContainer;
+  /** Application master specific info to register a new Application with RM/ASM */
+  private ApplicationSubmissionContext mAppContext;
+  /** Application name */
+  private String mAppName;
+  /** Application master priority */
+  private int mAmPriority;
+  /** Queue for App master */
+  private String mAmQueue;
+  /** Amount of memory resource to request for to run the App Master */
+  private int mAmMemory;
+  /** Number of virtual core resource to request for to run the App Master */
+  private int mAmVCores;
+  /** Application master jar file on HDFS */
+  private String mHDFSAppMasterJar;
+  /** Number of tachyon workers. */
+  private int mNumWorkers;
+  /** Tachyon home path on YARN containers. */
+  private String mTachyonHome;
+  /** Address to run Tachyon master. */
+  private String mMasterAddress;
+  /** Id of the application */
+  private ApplicationId mAppId;
+  /** Command line options */
+  private Options mOptions;
+
+  public Client() {
+    mOptions = new Options();
+    mOptions.addOption("appname", true, "Application Name. Default value Tachyon");
+    mOptions.addOption("priority", true, "Application Priority. Default 0");
+    mOptions.addOption("queue", true, "RM Queue in which this application is to be submitted");
+    mOptions.addOption("user", true, "User to run the application as");
+    mOptions.addOption("master_memory", true,
+        "Amount of memory in MB to be requested to run the application master");
+    mOptions.addOption("master_vcores", true, "Amount of virtual cores to be requested to run the "
+        + "application master");
+    mOptions.addOption("jar", true, "Jar file containing the application master");
+    mOptions.addOption("tachyon_home", true, "Path to Tachyon home dir on YARN slave machines");
+    mOptions.addOption("master_address", true, "Address to run Tachyon master");
+    mOptions.addOption("help", false, "Print usage");
+    mOptions.addOption("num_workers", true, "Number of tachyon workers to launch");
+  }
+
+  /**
+   * @param args Command line arguments
+   */
+  public static void main(String[] args) {
+    boolean result = false;
+    try {
+      Client client = new Client();
+      System.out.println("Initializing Client");
+      if (!client.parseArgs(args)) {
+        System.exit(0);
+      }
+      System.out.println("Starting Client");
+      result = client.run();
+    } catch (Throwable t) {
+      System.err.println("Error running Client " + t);
+      System.exit(1);
+    }
+    if (result) {
+      System.out.println("Application completed successfully");
+      System.exit(0);
+    }
+    System.err.println("Application failed to complete");
+    System.exit(2);
+  }
+
+  /**
+   * Main run function for the client
+   *
+   * @return true if application completed successfully
+   * @throws IOException
+   * @throws YarnException
```

Comment:
done---though it is very vague because that 's all I could get from YARN documentation.